core/bus/bus.go

```diff
@@ -67,12 +78,12 @@ func (b *Bus) Unregister(idx int) error {
 		return fmt.Errorf("could not unregister channel #%d: index out of range", idx)
 	}
 
+	b.current.connections -= 1
```

Comment:
Unregister should probably set some timer to delete the events / messages count from the metrics map, or else we technically have a small memory leak. I doubt that the system would ever get overtaken by an army of int64s, but... up to you how much you care about that.

Reply:
Event types are a finite set; the main problem we might run into is overflow, but the idea of monitoring this is to use something else (like RRD) to detect rollover and adjust the delta calculation. In that respect, the messages/bytes counters work more like packet/byte counters in network gear — ever-increasing.

Reply:
Missed that this is keying on the event type string and not the event object itself - which makes sense because you can't use a struct as a key.

Reply:
If overflow is a real concern, the `big` package is a thing.

Reply:
I'm really not.